### PR TITLE
ENH: integrate: support ILP64 in quadpack

### PR DESCRIPTION
**Pull request by pv**
#### Reference issue
Part of gh-11193

The first commit is gh-12090

#### What does this implement/fix?
Support 64-bit Fortran integers and BLAS in quadpack.
Again, the old hand-written C wrappers make the conversion more involved.

### CODE REVIEW

File: scipy/integrate/__quadpack.h [hunk 1]
```diff
@@ -521,23 +567,23 @@ static PyObject *quadpack_qagpe(PyObject *dummy, PyObject *args) {
...
+  iord = (F_INT *)ap_iord->data;
   alist = (double *)ap_alist->data;
   blist = (double *)ap_blist->data;
   rlist = (double *)ap_rlist->data;
   elist = (double *)ap_elist->data;
   pts = (double *)ap_pts->data;
-  level = (int *)ap_level->data;
-  ndin = (int *)ap_level->data;
+  level = (F_INT *)ap_level->data;
+  ndin = (F_INT *)ap_ndin->data;
```
Comment (pv) [inline scipy/integrate/__quadpack.h:586]: Note that the code in the old version seemed to have a bug here --- moreover, ap_level/ap_ndin arrays had the wrong types.

File: scipy/integrate/setup.py [hunk 2]
```diff
@@ -28,66 +34,73 @@ def configuration(parent_package='',top_path=None):
     quadpack_test_src = [join('tests','_test_multivariate.c')]
     odeint_banded_test_src = [join('tests', 'banded5x5.f')]
 
-    config.add_library('mach', sources=mach_src,
-                       config_fc={'noopt':(__file__,1)})
-    config.add_library('quadpack', sources=quadpack_src)
-    config.add_library('lsoda', sources=lsoda_src)
-    config.add_library('vode', sources=vode_src)
-    config.add_library('dop', sources=dop_src)
+    config.add_library('mach', sources=mach_src, config_fc={'noopt':(__file__,1)},
```
Comment (rgommers) [inline scipy/integrate/setup.py:37]: some more `lint_diff.py` pickiness: two spaces missing on this line

Comment (pv) [reply scipy/integrate/setup.py]: fixed

**Comment by rgommers** [comment]
LGTM, and MKL build is happy too. Merging, thanks @pv

**Pull request merged**